sql/catalog/lease: TestLeaseObserver failed

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/5ad489e7-c657-4cbb-bef8-1cb59fe0ebd4?testReportRun=3&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [8b6d3d2a14f67c0867d9f492fde1fad929538a51](https://github.com/cockroachdb/cockroach/commits/8b6d3d2a14f67c0867d9f492fde1fad929538a51):


```
I260417 04:26:56.764396 98979 server/server_controller_channel_orchestrator.go:263  [T1,Vsystem,n1] 709  server terminating; telling tenant "test-tenant" to terminate
W260417 04:26:56.764410 102950 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 710  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 04:26:56.764437 98916 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 711  error closing mux rangefeed client: context canceled
I260417 04:26:56.764503 99371 server/start_listen.go:114  [T10,Vtest-tenant,n1] 712  server shutting down: instructing cmux to stop accepting
W260417 04:26:56.764504 98782 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 713  error closing mux rangefeed client: context canceled
I260417 04:26:56.764525 99845 sql/stats/automatic_stats.go:766  [T10,Vtest-tenant,n1] 714  quiescing auto stats refresher
I260417 04:26:56.764537 99846 sql/stats/automatic_stats.go:804  [T10,Vtest-tenant,n1] 715  quiescing auto stats misestimates refresher
W260417 04:26:56.764604 99749 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620739593863169,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 716  error closing mux rangefeed client: context canceled
W260417 04:26:56.764629 98759 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 717  error closing mux rangefeed client: context canceled
I260417 04:26:56.764667 98745 sql/stats/automatic_stats.go:857  [T1,Vsystem,n1] 719  quiescing stats garbage collector
W260417 04:26:56.764665 102979 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 718  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 04:26:56.764689 99762 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620739593863169,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 720  error closing mux rangefeed client: context canceled
W260417 04:26:56.764712 103043 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 721  error closing mux rangefeed client: context canceled
W260417 04:26:56.764757 98507 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620738501050369,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 724  error closing mux rangefeed client: context canceled
W260417 04:26:56.764785 99477 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=sql_instances,mux_n=1,gen=2] 725  error closing mux rangefeed client: context canceled
W260417 04:26:56.764805 98493 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620738501050369,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 726  error closing mux rangefeed client: context canceled
W260417 04:26:56.764830 99393 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=lease,mux_n=1,gen=2] 727  error closing mux rangefeed client: context canceled
W260417 04:26:56.764845 99930 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 728  error closing mux rangefeed client: context canceled
W260417 04:26:56.764852 98193 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 729  error closing mux rangefeed client: context canceled
W260417 04:26:56.764900 99639 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 730  error closing mux rangefeed client: context canceled
W260417 04:26:56.764903 99656 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 732  error closing mux rangefeed client: context canceled
W260417 04:26:56.764942 97647 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 733  error closing mux rangefeed client: context canceled
W260417 04:26:56.764952 99727 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620739593863169,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 735  error closing mux rangefeed client: context canceled
W260417 04:26:56.764992 97654 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 736  error closing mux rangefeed client: context canceled
W260417 04:26:56.764730 99412 sql/sqlliveness/slinstance/slinstance.go:337  [T10,Vtest-tenant,n1] 723  exiting heartbeat loop
W260417 04:26:56.765168 99412 sql/sqlliveness/slinstance/slinstance.go:324  [T10,Vtest-tenant,n1] 739  exiting heartbeat loop with error: node unavailable; try another peer
E260417 04:26:56.765180 99412 server/server_sql.go:535  [T10,Vtest-tenant,n1] 740  failed to run update of instance with new session ID: node unavailable; try another peer
W260417 04:26:56.764949 99374 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vtest-tenant,n1,tenant-connector] 734  error consuming TenantSettings RPC: grpc: context canceled [code 1/Canceled]
I260417 04:26:56.765055 99560 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=1167620739593863169] 737  AUTO SPAN CONFIG RECONCILIATION job 1167620739593863169: stepping through state succeeded
W260417 04:26:56.765176 99804 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 741  error closing mux rangefeed client: context canceled
W260417 04:26:56.764712 98504 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167620738501050369,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 722  error closing mux rangefeed client: context canceled
W260417 04:26:56.765243 102896 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 742  could not clear job claim: clear-job-claim: node unavailable; try another peer
I260417 04:26:56.765252 98076 jobs/registry.go:1622  [T1,Vsystem,n1,job=1167620738501050369] 743  AUTO SPAN CONFIG RECONCILIATION job 1167620738501050369: stepping through state succeeded
I260417 04:26:56.765263 102927 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=103] 744  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W260417 04:26:56.765125 102931 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 738  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 04:26:56.764899 98176 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 731  error closing mux rangefeed client: context canceled
I260417 04:26:56.765315 99855 server/server_controller_channel_orchestrator.go:503  [n1,tenant-orchestration,tenant=test-tenant] 745  propagate-graceful-drain task terminating
I260417 04:26:56.765337 98980 server/server_controller_channel_orchestrator.go:443  [n1,tenant-orchestration,tenant=test-tenant] 746  tenant "test-tenant" finishing their own control loop
I260417 04:26:56.765351 99847 sql/stats/automatic_stats.go:857  [T10,Vtest-tenant,n1] 747  quiescing stats garbage collector
I260417 04:26:56.765410 99892 4@util/log/event_log.go:90  [T10,Vtest-tenant,n1,client=127.0.0.1:39486,hostssl,user=root] 748 ={"Timestamp":1776400016765407689,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:39486","SessionID":"18a70a5cc2b2e3fb0000000000000001","Duration":45224677346}
W260417 04:26:56.765404 99649 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 749  error closing mux rangefeed client: context canceled
I260417 04:26:56.770044 98981 1@util/log/event_log.go:90  [n1,tenant-orchestration,tenant=test-tenant] 750 ={"Timestamp":1776400016770042805,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"test-tenant"}
W260417 04:26:56.770070 98981 sql/event_log.go:620  [n1,tenant-orchestration,tenant=test-tenant] 751  failed to start task to save 1 events in eventlog: node unavailable; try another peer
W260417 04:26:56.770087 98981 server/server_sql.go:1847  [T10,Vtest-tenant,n1] 752  server shutdown without a prior graceful drain
I260417 04:26:56.770175 98981 server/server_controller_channel_orchestrator.go:487  [n1,tenant-orchestration,tenant=test-tenant] 753  propagate-ungraceful-stop task terminating
I260417 04:26:56.775249 98979 server/server_controller_channel_orchestrator.go:288  [T1,Vsystem,n1] 754  propagate-close task terminating
W260417 04:26:56.779814 96556 server/server_sql.go:1847  [T1,Vsystem,n1] 755  server shutdown without a prior graceful drain
I260417 04:26:56.779876 96556 server/server_controller.go:380  [T1,Vsystem,n1] 756  server controller shutting down
I260417 04:26:56.779891 96556 server/server_controller.go:389  [T1,Vsystem,n1] 757  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (45.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63003

sql/catalog/lease.TestLeaseObserver [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21314772?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21314772?buildTab=artifacts#/) on master @ [8b6d3d2a14f67c0867d9f492fde1fad929538a51](https://github.com/cockroachdb/cockroach/commits/8b6d3d2a14f67c0867d9f492fde1fad929538a51):


```
W260417 03:35:50.645631 134040 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 726  error closing mux rangefeed client: context canceled
W260417 03:35:50.645695 130417 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 727  error closing mux rangefeed client: context canceled
W260417 03:35:50.645800 130721 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 728  error closing mux rangefeed client: context canceled
W260417 03:35:50.645850 130679 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167610692278517761,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 729  error closing mux rangefeed client: context canceled
W260417 03:35:50.646042 133871 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 730  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 03:35:50.646044 130676 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167610692278517761,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 731  error closing mux rangefeed client: context canceled
I260417 03:35:50.645804 130280 server/start_listen.go:114  [T10,Vtest-tenant,n1] 732  server shutting down: instructing cmux to stop accepting
I260417 03:35:50.646196 127714 server/start_listen.go:114  [T1,Vsystem,n1] 733  server shutting down: instructing cmux to stop accepting
W260417 03:35:50.646237 130662 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167610692278517761,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 734  error closing mux rangefeed client: context canceled
W260417 03:35:50.646392 130522 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 735  error closing mux rangefeed client: context canceled
W260417 03:35:50.646430 129698 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 736  error closing mux rangefeed client: context canceled
W260417 03:35:50.646547 129268 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 740  error closing mux rangefeed client: context canceled
W260417 03:35:50.646652 130357 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=lease,mux_n=1,gen=2] 741  error closing mux rangefeed client: context canceled
W260417 03:35:50.646668 128524 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=sql_instances,mux_n=1,gen=2] 742  error closing mux rangefeed client: context canceled
W260417 03:35:50.646676 128437 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 744  error closing mux rangefeed client: context canceled
W260417 03:35:50.646765 129358 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167610690730229761,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 748  error closing mux rangefeed client: context canceled
W260417 03:35:50.646439 131822 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 737  could not clear job claim: clear-job-claim: node unavailable; try another peer
I260417 03:35:50.646436 130737 sql/stats/automatic_stats.go:766  [T10,Vtest-tenant,n1] 738  quiescing auto stats refresher
I260417 03:35:50.646457 130738 sql/stats/automatic_stats.go:804  [T10,Vtest-tenant,n1] 739  quiescing auto stats misestimates refresher
W260417 03:35:50.646627 130283 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vtest-tenant,n1,tenant-connector] 743  error consuming TenantSettings RPC: grpc: context canceled [code 1/Canceled]
W260417 03:35:50.646698 130327 sql/sqlliveness/slinstance/slinstance.go:337  [T10,Vtest-tenant,n1] 745  exiting heartbeat loop
W260417 03:35:50.646867 130327 sql/sqlliveness/slinstance/slinstance.go:324  [T10,Vtest-tenant,n1] 750  exiting heartbeat loop with error: node unavailable; try another peer
E260417 03:35:50.646883 130327 server/server_sql.go:535  [T10,Vtest-tenant,n1] 752  failed to run update of instance with new session ID: node unavailable; try another peer
W260417 03:35:50.646881 129352 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167610690730229761,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 751  error closing mux rangefeed client: context canceled
I260417 03:35:50.646743 130475 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=1167610692278517761] 747  AUTO SPAN CONFIG RECONCILIATION job 1167610692278517761: stepping through state succeeded
I260417 03:35:50.646918 128779 jobs/registry.go:1622  [T1,Vsystem,n1,job=1167610690730229761] 753  AUTO SPAN CONFIG RECONCILIATION job 1167610690730229761: stepping through state succeeded
W260417 03:35:50.646735 130282 kv/kvclient/kvtenant/connector.go:397  [T10,Vtest-tenant,n1,tenant-connector] 746  error consuming GossipSubscription RPC: grpc: context canceled [code 1/Canceled]
W260417 03:35:50.647021 133885 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 754  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 03:35:50.647081 130704 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 755  error closing mux rangefeed client: context canceled
W260417 03:35:50.647165 128779 jobs/adopt.go:563  [T1,Vsystem,n1] 756  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 03:35:50.647283 129577 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 757  error closing mux rangefeed client: context canceled
W260417 03:35:50.647385 128502 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 758  error closing mux rangefeed client: context canceled
W260417 03:35:50.647395 127985 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 759  exiting heartbeat loop
W260417 03:35:50.647413 127985 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 760  exiting heartbeat loop with error: node unavailable; try another peer
E260417 03:35:50.647427 127985 server/server_sql.go:535  [T1,Vsystem,n1] 761  failed to run update of instance with new session ID: node unavailable; try another peer
I260417 03:35:50.646796 133820 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=103] 749  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W260417 03:35:50.647461 129788 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 762  error closing mux rangefeed client: context canceled
I260417 03:35:50.647483 130775 4@util/log/event_log.go:90  [T10,Vtest-tenant,n1,client=127.0.0.1:35386,hostssl,user=root] 763 ={"Timestamp":1776396950647480145,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:35386","SessionID":"18a70792dd2893390000000000000001","Duration":45269480354}
W260417 03:35:50.647495 133702 jobs/adopt.go:563  [T1,Vsystem,n1] 764  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260417 03:35:50.647550 127732 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 765  error closing mux rangefeed client: context canceled
W260417 03:35:50.647612 133814 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 766  error closing mux rangefeed client: context canceled
I260417 03:35:50.650174 129799 1@util/log/event_log.go:90  [n1,tenant-orchestration,tenant=test-tenant] 767 ={"Timestamp":1776396950650172586,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"test-tenant"}
W260417 03:35:50.650207 129799 sql/event_log.go:620  [n1,tenant-orchestration,tenant=test-tenant] 768  failed to start task to save 1 events in eventlog: node unavailable; try another peer
W260417 03:35:50.650228 129799 server/server_sql.go:1847  [T10,Vtest-tenant,n1] 769  server shutdown without a prior graceful drain
I260417 03:35:50.650383 129799 server/server_controller_channel_orchestrator.go:487  [n1,tenant-orchestration,tenant=test-tenant] 770  propagate-ungraceful-stop task terminating
I260417 03:35:50.651901 129797 server/server_controller_channel_orchestrator.go:288  [T1,Vsystem,n1] 771  propagate-close task terminating
W260417 03:35:50.656347 127099 server/server_sql.go:1847  [T1,Vsystem,n1] 772  server shutdown without a prior graceful drain
I260417 03:35:50.656473 127099 server/server_controller.go:380  [T1,Vsystem,n1] 773  server controller shutting down
I260417 03:35:50.656490 127099 server/server_controller.go:389  [T1,Vsystem,n1] 774  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (46.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/ee0d1576-89b8-44de-88f4-b0945c212ab8?testReportRun=3&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [ad0ff8264462b3d5bc0b34db272939c8c17999bb](https://github.com/cockroachdb/cockroach/commits/ad0ff8264462b3d5bc0b34db272939c8c17999bb):


```
W260418 04:23:14.240352 117263 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 710  error closing mux rangefeed client: context canceled
W260418 04:23:14.240382 120971 jobs/adopt.go:563  [T1,Vsystem,n1] 711  could not clear job claim: clear-job-claim: node unavailable; try another peer
I260418 04:23:14.240410 117130 sql/stats/automatic_stats.go:804  [T1,Vsystem,n1] 712  quiescing auto stats misestimates refresher
W260418 04:23:14.240435 116897 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903124577058817,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 713  error closing mux rangefeed client: context canceled
W260418 04:23:14.240465 121101 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 714  error closing mux rangefeed client: context canceled
W260418 04:23:14.240512 117100 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 715  error closing mux rangefeed client: context canceled
W260418 04:23:14.240558 121470 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 716  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260418 04:23:14.240551 116899 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903124577058817,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 717  error closing mux rangefeed client: context canceled
I260418 04:23:14.240578 121500 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=103] 718  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W260418 04:23:14.240590 116894 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903124577058817,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 719  error closing mux rangefeed client: context canceled
I260418 04:23:14.240631 116487 jobs/registry.go:1622  [T1,Vsystem,n1,job=1167903124577058817] 720  AUTO SPAN CONFIG RECONCILIATION job 1167903124577058817: stepping through state succeeded
W260418 04:23:14.240628 116561 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 721  error closing mux rangefeed client: context canceled
W260418 04:23:14.240664 115582 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 722  exiting heartbeat loop
W260418 04:23:14.240673 115582 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 723  exiting heartbeat loop with error: node unavailable; try another peer
E260418 04:23:14.240684 115582 server/server_sql.go:535  [T1,Vsystem,n1] 724  failed to run update of instance with new session ID: node unavailable; try another peer
W260418 04:23:14.240748 121500 jobs/adopt.go:563  [T10,Vtest-tenant,n1] 725  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260418 04:23:14.240742 117808 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 726  error closing mux rangefeed client: context canceled
I260418 04:23:14.240806 118245 server/server_controller_channel_orchestrator.go:503  [n1,tenant-orchestration,tenant=test-tenant] 727  propagate-graceful-drain task terminating
I260418 04:23:14.240830 118205 sql/stats/automatic_stats.go:857  [T10,Vtest-tenant,n1] 728  quiescing stats garbage collector
I260418 04:23:14.240856 117297 server/server_controller_channel_orchestrator.go:443  [n1,tenant-orchestration,tenant=test-tenant] 729  tenant "test-tenant" finishing their own control loop
W260418 04:23:14.240859 117854 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=sql_instances,mux_n=1,gen=2] 730  error closing mux rangefeed client: context canceled
W260418 04:23:14.240897 117818 sql/sqlliveness/slinstance/slinstance.go:337  [T10,Vtest-tenant,n1] 731  exiting heartbeat loop
W260418 04:23:14.240908 117818 sql/sqlliveness/slinstance/slinstance.go:324  [T10,Vtest-tenant,n1] 732  exiting heartbeat loop with error: node unavailable; try another peer
E260418 04:23:14.240918 117818 server/server_sql.go:535  [T10,Vtest-tenant,n1] 733  failed to run update of instance with new session ID: node unavailable; try another peer
W260418 04:23:14.240962 118129 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903125817688065,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 734  error closing mux rangefeed client: context canceled
W260418 04:23:14.240911 117366 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 735  error closing mux rangefeed client: context canceled
I260418 04:23:14.240990 117787 server/start_listen.go:114  [T10,Vtest-tenant,n1] 736  server shutting down: instructing cmux to stop accepting
W260418 04:23:14.241044 117790 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vtest-tenant,n1,tenant-connector] 737  error consuming TenantSettings RPC: grpc: node unavailable; try another peer [code 2/Unknown]
W260418 04:23:14.241108 118163 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903125817688065,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 738  error closing mux rangefeed client: context canceled
W260418 04:23:14.241123 118357 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 739  error closing mux rangefeed client: context canceled
I260418 04:23:14.241169 118203 sql/stats/automatic_stats.go:766  [T10,Vtest-tenant,n1] 740  quiescing auto stats refresher
I260418 04:23:14.241182 118204 sql/stats/automatic_stats.go:804  [T10,Vtest-tenant,n1] 741  quiescing auto stats misestimates refresher
W260418 04:23:14.241172 118096 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1167903125817688065,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 742  error closing mux rangefeed client: context canceled
I260418 04:23:14.241206 117905 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=1167903125817688065] 743  AUTO SPAN CONFIG RECONCILIATION job 1167903125817688065: stepping through state succeeded
W260418 04:23:14.241225 118066 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 744  error closing mux rangefeed client: context canceled
W260418 04:23:14.241268 117789 kv/kvclient/kvtenant/connector.go:397  [T10,Vtest-tenant,n1,tenant-connector] 745  error consuming GossipSubscription RPC: grpc: node unavailable; try another peer [code 2/Unknown]
W260418 04:23:14.241350 118272 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 746  error closing mux rangefeed client: context canceled
W260418 04:23:14.241392 121605 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 747  error closing mux rangefeed client: context canceled
W260418 04:23:14.241391 117846 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=lease,mux_n=1,gen=2] 748  error closing mux rangefeed client: context canceled
W260418 04:23:14.241490 118055 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 749  error closing mux rangefeed client: context canceled
W260418 04:23:14.241617 117180 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 750  error closing mux rangefeed client: context canceled
I260418 04:23:14.245196 117378 1@util/log/event_log.go:90  [n1,tenant-orchestration,tenant=test-tenant] 751 ={"Timestamp":1776486194245193659,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"test-tenant"}
W260418 04:23:14.245236 117378 sql/event_log.go:620  [n1,tenant-orchestration,tenant=test-tenant] 752  failed to start task to save 1 events in eventlog: node unavailable; try another peer
W260418 04:23:14.245258 117378 server/server_sql.go:1847  [T10,Vtest-tenant,n1] 753  server shutdown without a prior graceful drain
I260418 04:23:14.245393 117378 server/server_controller_channel_orchestrator.go:487  [n1,tenant-orchestration,tenant=test-tenant] 754  propagate-ungraceful-stop task terminating
I260418 04:23:14.250327 117296 server/server_controller_channel_orchestrator.go:288  [T1,Vsystem,n1] 755  propagate-close task terminating
W260418 04:23:14.256122 114980 server/server_sql.go:1847  [T1,Vsystem,n1] 756  server shutdown without a prior graceful drain
I260418 04:23:14.256238 114980 server/server_controller.go:380  [T1,Vsystem,n1] 757  server controller shutting down
I260418 04:23:14.256254 114980 server/server_controller.go:389  [T1,Vsystem,n1] 758  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (45.89s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/4d85ce4b-b96a-4b1a-8027-527df103d60b?testReportRun=8&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [a41e86dbb2bd18447a4ccf69d6c5f887827cd7d7](https://github.com/cockroachdb/cockroach/commits/a41e86dbb2bd18447a4ccf69d6c5f887827cd7d7):


```
I260419 04:20:00.018886 115719 testutils/soon.go:38  [-] 417  SucceedsSoon: new versions were not detected
I260419 04:20:01.019011 115719 testutils/soon.go:38  [-] 418  SucceedsSoon: new versions were not detected
I260419 04:20:02.020185 115719 testutils/soon.go:38  [-] 419  SucceedsSoon: new versions were not detected
I260419 04:20:03.021138 115719 testutils/soon.go:38  [-] 420  SucceedsSoon: new versions were not detected
I260419 04:20:04.021512 115719 testutils/soon.go:38  [-] 421  SucceedsSoon: new versions were not detected
I260419 04:20:05.022573 115719 testutils/soon.go:38  [-] 422  SucceedsSoon: new versions were not detected
I260419 04:20:06.023378 115719 testutils/soon.go:38  [-] 423  SucceedsSoon: new versions were not detected
I260419 04:20:07.023736 115719 testutils/soon.go:38  [-] 424  SucceedsSoon: new versions were not detected
I260419 04:20:07.507644 116285 2@util/log/event_log.go:90  [T1,Vsystem,n1] 425 ={"Timestamp":1776572407507639203,"EventType":"runtime_stats","MemRSSBytes":1069699072,"GoroutineCount":368,"MemStackSysBytes":10846208,"GoAllocBytes":681096352,"GoTotalBytes":974197256,"HeapFragmentBytes":133073760,"HeapReservedBytes":126107648,"HeapReleasedBytes":84869120,"CGoAllocBytes":18948400,"CGoTotalBytes":65941504,"CGoCallRate":0.19999526,"CPUUserPercent":3.8999076,"CPUSysPercent":3.3999197,"GCRunCount":47,"NetHostRecvBytes":208,"NetHostSendBytes":208}
I260419 04:20:07.507757 116285 2@server/status/runtime_log.go:43  [T1,Vsystem,n1] 426  runtime stats: 1020 MiB RSS, 368 goroutines (stacks: 10 MiB), 650 MiB/929 MiB Go alloc/total (heap fragmentation: 127 MiB, heap reserved: 120 MiB, heap released: 81 MiB), 18 MiB/63 MiB CGO alloc/total (0.2 CGO/sec), 3.9/3.4 %(u/s)time, 0.0 %gc (47x), 208 B/208 B (r/w)net
I260419 04:20:08.024909 115719 testutils/soon.go:38  [-] 427  SucceedsSoon: new versions were not detected
I260419 04:20:09.025016 115719 testutils/soon.go:38  [-] 428  SucceedsSoon: new versions were not detected
I260419 04:20:10.026117 115719 testutils/soon.go:38  [-] 429  SucceedsSoon: new versions were not detected
I260419 04:20:11.027085 115719 testutils/soon.go:38  [-] 430  SucceedsSoon: new versions were not detected
I260419 04:20:12.028025 115719 testutils/soon.go:38  [-] 431  SucceedsSoon: new versions were not detected
W260419 04:20:12.509989 116310 util/admission/store_grant_coordinator.go:184  [T1,Vsystem,n1] 432  expected 1 disk stats and found 0 stats
    observer_test.go:125: condition failed to evaluate within 45s: from observer_test.go:127: new versions were not detected
        
        goroutine dump: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/sql/catalog/lease/lease_test/shard_3_of_4_run_8_of_10/test.outputs/goroutine_dump_1981669319.txt
I260419 04:20:13.032946 118938 jobs/registry.go:1622  [T1,Vsystem,n1,job=100] 433  KEY VISUALIZER job 100: stepping through state succeeded
W260419 04:20:13.033260 117916 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 437  error closing mux rangefeed client: context canceled
W260419 04:20:13.033290 118016 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 438  error closing mux rangefeed client: context canceled
W260419 04:20:13.033370 117600 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1168185647206858753,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 439  error closing mux rangefeed client: context canceled
I260419 04:20:13.033413 117866 sql/stats/automatic_stats.go:766  [T1,Vsystem,n1] 440  quiescing auto stats refresher
I260419 04:20:13.033427 117867 sql/stats/automatic_stats.go:804  [T1,Vsystem,n1] 441  quiescing auto stats misestimates refresher
W260419 04:20:13.033413 118033 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 442  error closing mux rangefeed client: context canceled
W260419 04:20:13.033496 117242 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 443  error closing mux rangefeed client: context canceled
W260419 04:20:13.033579 117196 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 444  error closing mux rangefeed client: context canceled
W260419 04:20:13.033710 118881 jobs/adopt.go:563  [T1,Vsystem,n1] 445  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260419 04:20:13.033730 116709 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 446  exiting heartbeat loop
W260419 04:20:13.033738 116709 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 447  exiting heartbeat loop with error: node unavailable; try another peer
E260419 04:20:13.033750 116709 server/server_sql.go:535  [T1,Vsystem,n1] 448  failed to run update of instance with new session ID: node unavailable; try another peer
W260419 04:20:13.033816 118948 jobs/adopt.go:563  [T1,Vsystem,n1] 449  could not clear job claim: clear-job-claim: node unavailable; try another peer
W260419 04:20:13.033882 116490 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 450  error closing mux rangefeed client: context canceled
W260419 04:20:13.033965 118024 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 451  error closing mux rangefeed client: context canceled
W260419 04:20:13.033972 117277 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 452  error closing mux rangefeed client: context canceled
W260419 04:20:13.034050 117653 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1168185647206858753,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 453  error closing mux rangefeed client: context canceled
W260419 04:20:13.034138 117825 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 454  error closing mux rangefeed client: context canceled
W260419 04:20:13.034127 116747 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 455  error closing mux rangefeed client: context canceled
I260419 04:20:13.034185 118924 jobs/registry.go:1622  [T1,Vsystem,n1,job=103] 456  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W260419 04:20:13.034197 117650 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1168185647206858753,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 457  error closing mux rangefeed client: context canceled
I260419 04:20:13.034231 117192 jobs/registry.go:1622  [T1,Vsystem,n1,job=1168185647206858753] 458  AUTO SPAN CONFIG RECONCILIATION job 1168185647206858753: stepping through state succeeded
W260419 04:20:13.034309 119050 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=POLL JOBS STATS id=101,rangefeed=cluster-metrics-watcher,mux_n=1,gen=2] 459  error closing mux rangefeed client: context canceled
I260419 04:20:13.033132 117936 4@util/log/event_log.go:90  [T1,Vsystem,n1,client=127.0.0.1:59474,hostssl,user=root] 435 ={"Timestamp":1776572413033129699,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:59474","SessionID":"18a7a727e2b676520000000000000001","Duration":45262999980}
I260419 04:20:13.033058 117868 sql/stats/automatic_stats.go:857  [T1,Vsystem,n1] 434  quiescing stats garbage collector
I260419 04:20:13.033161 116200 server/start_listen.go:114  [T1,Vsystem,n1] 436  server shutting down: instructing cmux to stop accepting
W260419 04:20:13.038313 115719 server/server_sql.go:1847  [T1,Vsystem,n1] 460  server shutdown without a prior graceful drain
I260419 04:20:13.038390 115719 server/server_controller.go:380  [T1,Vsystem,n1] 461  server controller shutting down
I260419 04:20:13.038412 115719 server/server_controller.go:389  [T1,Vsystem,n1] 462  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (45.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/lease/observer_test.go
+++ b/pkg/sql/catalog/lease/observer_test.go
@@ -75,9 +75,15 @@ func TestLeaseObserver(t *testing.T) {
 	defer maybeCloseSchemaChangeComplete()
 	var maxVersion atomic.Int64
 	maxVersion.Store(1)
+	var grpErr atomic.Pointer[error]
 	// Ensure the first version is locked before we start the schema change.
 	firstVersionLocked := make(chan struct{})
-	grp.GoCtx(func(ctx context.Context) error {
+	grp.GoCtx(func(ctx context.Context) (retErr error) {
+		defer func() {
+			if retErr != nil {
+				grpErr.Store(&retErr)
+			}
+		}()
 		firstVersionLockedClosed := false
 		maybeCloseFirstVersionLocked := func() {
 			if firstVersionLockedClosed {
@@ -99,10 +105,9 @@ func TestLeaseObserver(t *testing.T) {
 		for {
 			select {
 			case v := <-obs.notified:
-				if ld.Underlying().GetVersion()+1 != v {
-					return errors.AssertionFailedf("expected version %d, got %d", ld.Underlying().GetVersion()+1, v)
+				if v <= ld.Underlying().GetVersion() {
+					continue
 				}
-				require.Equal(t, ld.Underlying().GetVersion()+1, v)
 				maxVersion.Store(int64(v))
 				newLease, err := lm.Acquire(ctx, lease.TimestampToReadTimestamp(kvDB.Clock().Now()), tableID)
 				if err != nil {
@@ -110,9 +115,12 @@ func TestLeaseObserver(t *testing.T) {
 				}
 				ld.Release(ctx)
 				ld = newLease
-				if ld.Underlying().GetVersion() != descpb.DescriptorVersion(maxVersion.Load()) {
-					return errors.AssertionFailedf("expected version %d, got %d", maxVersion.Load(), ld.Underlying().GetVersion())
+				if ld.Underlying().GetVersion() < descpb.DescriptorVersion(maxVersion.Load()) {
+					return errors.AssertionFailedf(
+						"expected version >= %d, got %d", maxVersion.Load(), ld.Underlying().GetVersion(),
+					)
 				}
+				maxVersion.Store(int64(ld.Underlying().GetVersion()))
 			case <-schemaChangeComplete:
 				return nil
 			}
@@ -123,6 +131,9 @@ func TestLeaseObserver(t *testing.T) {
 	sqlRunner.Exec(t, "ALTER TABLE t ADD COLUMN v INT")
 	// Validate the schema change was observed.
 	testutils.SucceedsSoon(t, func() error {
+		if e := grpErr.Load(); e != nil {
+			return *e
+		}
 		if maxVersion.Load() <= int64(1) {
 			return errors.New("new versions were not detected")
 		}

